elastic/kibana
# [Ingest pipelines] Upload indexed document to test a pipeline

This PR adds the ability to upload a document(s) from an existing index when testing a pipeline.

Fixes https://github.com/elastic/kibana/issues/74796

Skipping release note as I think this change can fall under the general improvements made to the debugging workflow. Release note added on https://github.com/elastic/kibana/pull/74964.

### How to review

#### Testing instructions
1. Create a pipeline (for a sample pipeline, see: https://github.com/elastic/kibana/pull/74964).
2. Create an index with at least one document, or use the kibana sample data. For example:
```
PUT /test/_doc/1
{
  "name": "John Doe"
}
```
3. Click the "Add documents" link
4. Add a sample document in the "Documents" code editor (you can just use the example in the help text)
5. Click the "Import existing documents" accordion
6. Add the index name and document ID of the document from step 2 and click "Import". Note: You can use the link to "Discover" to help find the document to use.
7. Verify the document was added successfully, the document added in step 4 still exists,  and you are able to run the pipeline.

#### UX
I'm interested in any design feedback on this. Two questions in particular on my end:

- Does it make sense to have a single form to import documents (particularly in the case where you might want to add more than one document)? I first considered doing something similar to what we see in the mappings editor for the `relationships` parameter, where you could add multiple rows, but I'm not sure how useful this is.

Example from mappings editor:
<img width="699" alt="Screen Shot 2020-09-22 at 1 37 31 PM" src="https://user-images.githubusercontent.com/5226211/93917436-ff9f8000-fcd8-11ea-92af-16ff14c04254.png">

- Thoughts on having the imported document populate the code editor below? I think we had initially discussed potentially having some sort of preview of the document you selected. However, I thought adding it to the code editor might be nice in the scenario where you might want to tweak the imported document, or also add a second sample document with JSON.


### Screenshots

Initial view:
<img width="557" alt="Screen Shot 2020-09-21 at 3 11 56 PM" src="https://user-images.githubusercontent.com/5226211/93813462-2601e480-fc21-11ea-8ea8-a3b53bfcc55a.png">

Accordion open:
<img width="561" alt="Screen Shot 2020-09-21 at 3 12 55 PM" src="https://user-images.githubusercontent.com/5226211/93813406-12567e00-fc21-11ea-8637-cb771430fac7.png">

Field errors:
<img width="566" alt="Screen Shot 2020-09-21 at 3 13 00 PM" src="https://user-images.githubusercontent.com/5226211/93813409-12ef1480-fc21-11ea-81f3-b98c74235033.png">

API error:
<img width="598" alt="Screen Shot 2020-09-21 at 3 13 10 PM" src="https://user-images.githubusercontent.com/5226211/93813412-12ef1480-fc21-11ea-96b1-73aae192ec38.png">
